pkg/utils/utils.go

```diff
@@ -66,12 +66,19 @@ func GetPfName(pciAddr string) (string, error) {
 			path := filepath.Join(sysBusPci, pciAddr, "net")
 			files, err = ioutil.ReadDir(path)
 			if err == nil {
-				return files[0].Name(), nil
+				if len(files) > 0 {
```

Comment:
The extra nesting could be avoided to improve readability, by handling the errors first.
That would match the go tendency to have the happy path on the leftmost side.

```
   if err != nil {
       return "", err
   }
   if len(files) <= 0 {
       return "", fmt.Errorf("no interface name found for device %s", pciAddr)
   }
   return files[0].Name(), nil
```